docs/topics/development/troubleshooting_and_debugging.md

```diff
@@ -0,0 +1,176 @@
+# Troubleshooting and Debugging
+
+Effective troubleshooting and debugging practices are essential for maintaining and improving the **addons-server** project. This section covers common issues, their solutions, and tools for effective debugging.
+
+## Common Issues and Solutions
+
+1. **Containers Not Starting**:
+   - **Issue**: Docker containers fail to start.
+   - **Solution**: Ensure that Docker is running and that no other services are using the required ports. Use the following command to start the containers:
+
+     ```sh
+     make up
+     ```
+
+2. **Database Connection Errors**:
+   - **Issue**: The application cannot connect to the MySQL database.
+   - **Solution**: Verify that the MySQL container is running and that the connection details in the `.env` file are correct. Restart the MySQL container if necessary:
+
+     ```sh
+     make down
+     make up
+     ```
+
+3. **Missing Dependencies**:
+   - **Issue**: Missing Python or Node.js dependencies.
+   - **Solution**: Ensure that all dependencies are installed by running the following command:
+
+     ```sh
+     make update_deps
+     ```
+
+4. **Locale Compilation Issues**:
+   - **Issue**: Locales are not compiling correctly.
+   - **Solution**: Run the locale compilation command and check for any errors:
+
+     ```sh
+     make compile_locales
+     ```
+
+5. **Permission Issues**:
+   - **Issue**: Permission errors when accessing files or directories.
+   - **Solution**: Ensure that the `olympia` user has the correct permissions. Use `chown` or `chmod` to adjust permissions if necessary.
+
+## Debugging
+
+The Docker setup uses `supervisord` to run the Django runserver. This allows you to access the management server from a shell to run things like `ipdb`.
+
+### Using `ipdb`
```

Comment:
mention using ` -x --pdbcls=IPython.core.debugger:Pdb --pdb` with pytest to fall into a debugger when there's a test exception?

Reply:
Can you clarify exactly what change you want? It's not clear to me exactly what to do here.

Reply:
Just that we're mentioning debugging, and (for me, at least) a fairly common case is debugging why something is failing during a test.  I'm suggesting we add a few lines informing/reminding that you can launch a debugger during a test at the point there's an exception or failure if pytest is executed with those parameters.
e.g. `pytest src/olympia/addons/tests/test_models.py::TestAddon -x --pdbcls=IPython.core.debugger:Pdb --pdb` which would start a debugger at the point there was an exception rather than just failing the test.

Reply:
wdyt?